node/src/BaseClient.ts

```diff
@@ -348,20 +348,63 @@ function convertGlideRecordForZSet(
 
 /**
  * @internal
- * Downcast `GlideRecord` to `Record`. Use if you are 146% aware that `data` keys are always strings.
+ * Recursevly downcast `GlideRecord` to `Record`. Use if `data` keys are always strings.
  */
 export function glideRecordToRecord<T>(
     data: GlideRecord<T>,
 ): Record<string, T> {
     const res: Record<string, T> = {};
 
     for (const pair of data) {
-        res[pair.key as string] = pair.value;
+        let newVal = pair.value;
```

Comment:
consider: 
```
        const value = pair.value;
        let newVal = 
            isGlideRecord(value) ? glideRecordToRecord(value as GlideRecord<unknown>) as T :
            isGlideRecordArray(value) ? (value as GlideRecord<unknown>[]).map(glideRecordToRecord) as T :
            value;
```

Reply:
you could also use a 
```
const res = data.map(...)
```

Reply:
Let me avoid that. linter makes ternary operators unreadable